src/Wangkanai.Detection/Version.cs

```diff
@@ -1,18 +1,26 @@
 // Copyright (c) 2016 Sarin Na Wangkanai, All Rights Reserved.
 // The GNU GPLv3. See License.txt in the project root for license information.
 
+using System;
+
 namespace Wangkanai.Detection
 {
     public class Version
     {
         public Version(string major, string minor)
         {
+            if (major == null) throw new ArgumentNullException(nameof(major));
+            if (minor == null) throw new ArgumentNullException(nameof(minor));
+
             Major = major;
             Minor = minor;
         }
         public Version(string major, string minor, string patch, string build)
             :this(major, minor)
         {
+            if (patch == null) throw new ArgumentNullException(nameof(patch));
+            if (build == null) throw new ArgumentNullException(nameof(build));
```

Comment:
If we check `patch` and `build` for `null` we have to update `Ctor_StringStringStringString_Success()` test inline data and remove the null values.